### PR TITLE
Updated deps for solving ocb conflicts

### PR DESCRIPTION
**Pull request by harisudarsan1**
The following changes fixes the Dockerfile and updates the dependencies to solve conflicts with the opentelementry custom builder.

### CODE REVIEW

File: kubearmor_receiver/go.mod [hunk 1]
```diff
@@ -1,57 +1,66 @@
 module github.com/kubearmor/OTel-receiver/kubearmor_receiver
 
-go 1.20
+go 1.21
```
Comment (DelusionalOptimist) [inline kubearmor_receiver/go.mod:3]: Can we use `go 1.22` with `toolchain go1.22.1`
Go 1.21 had some vulnerabilities. You can refer to https://github.com/kubearmor/kubearmor-client/pull/414 for the same.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.20-alpine3.17 as builder
+FROM golang:alpine3.19 as builder
```
Comment (DelusionalOptimist) [inline Dockerfile:1]: The above will always fetch the latest major version of Go which might lead to unexpected failures. Let's use something like below:
```suggestion
FROM golang:1.22-alpine3.19 as builder
```

**Review by DelusionalOptimist — COMMENTED** [review]
Looks mostly good. Some nit picks, then we are good to merge.
Can you also update Go version in [renovate.json](https://github.com/kubearmor/otel-adapter/blob/main/renovate.json) and [CI](https://github.com/kubearmor/otel-adapter/blob/main/.github/workflows/ci-test.yml)?

**Comment by harisudarsan1** [comment]
@DelusionalOptimist Added the requested changes

**Review by DelusionalOptimist — COMMENTED** [review]
LGTM. Thanks @harisudarsan1! :rocket:

**Pull request merged**